gnosis/gp-swap-ui
# 563/fix eth sign v4 signature not supported

# Update 2021-05-27

- Using new version of contracts NPM lib
- No longer using my hacked version of ethers.js lib
- This change fixes 2 wallets:
    - TrustWallet mobile browser
    - Opera browser wallet (Fantom)
- Trying different signing methods in sequence:
    - v4 (default and preferred)
    - v3 (when v4 not available)
    - eth_sign (when v3 not available)

# Summary

Closes #563 

After this change, I'm able to sign orders on Trust wallet dapp browser.
Need to test with more wallets of this type (injected)

# Testing
1. Using Trust wallet mobile dapp browser, open this PR's deployed app
2. Place an order like you normally do (fill in all fields, allow a token if needed)
- [ ] A signature request of a typed data should be requested
- [ ] The signature should succeed and the order should be placed successfully

3. Using Opera browser, setup the built in crypto wallet
4. In your mobile device, install Opera and setup crypto wallet there as well
4. Open this PR's deployed app
5. Place an order
- [ ] On your phone, you should see the signature request of the hash
- [ ] The signature should succeed and the order should be placed successfully
